src/components/core/geoChart/GeoChartLegendRenderer.tsx

```diff
@@ -1,216 +1,84 @@
     execution: Execution.IExecutionResponses;
     locale: string;
     position?: string;
 }
-export default class GeoChartLegendRenderer extends React.PureComponent<IChartLegendProps> {
-    constructor(props: IChartLegendProps) {
-        super(props);
-    }
-
+export default class GeoChartLegendRenderer extends React.PureComponent<IGeoChartLegendRendererProps> {
+    public static defaultProps: Partial<IGeoChartLegendRendererProps> = {
+        config: {},
+    };
     public render() {
         const {
-            execution: { executionResult },
+            execution: { executionResult, executionResponse },
             config: { mdObject: { buckets = [] } = {} },
             locale,
             position = TOP,
         } = this.props;
 
-        const format: string = "#,##0.00"; // TODO: will implement with SegmentBy or Color Legend
         const geoDataIndex: IGeoDataIndex = getGeoDataIndex(buckets);
         const data = executionResult.data as Execution.DataValue[][];
         const classes = ["geo-legend", `position-${position}`];
         return (
             <div className={classNames(...classes)}>
-                {(geoDataIndex.hasOwnProperty(COLOR) || geoDataIndex.hasOwnProperty(SEGMENT_BY)) &&
-                    this.renderPushpinColorOrSegmentByLegend(geoDataIndex.hasOwnProperty(SEGMENT_BY))}
-                {geoDataIndex.hasOwnProperty(SIZE) && (
-                    <IntlWrapper locale={locale}>
-                        <IntlTranslationsProvider>
-                            {(props: ITranslationsComponentProps) =>
-                                this.renderPushpinSizeLegend(
-                                    (data[geoDataIndex.size] || []).map(stringToFloat),
-                                    format,
-                                    props.numericSymbols,
-                                )
-                            }
-                        </IntlTranslationsProvider>
-                    </IntlWrapper>
-                )}
+                {geoDataIndex.hasOwnProperty(SIZE) &&
+                    this.renderPushpinSizeLegend(
+                        (data[geoDataIndex.size] || []).map(stringToFloat),
+                        this.getFormatFromExecution(geoDataIndex.size, executionResponse),
+                        locale,
+                    )}
             </div>
         );
     }
 
-    private renderPushpinSizeLegend(sizeValues: number[], format: string, numericSymbols: string[]) {
+    private getFormatFromExecution(indexMeasure: number, result: Execution.IExecutionResponse): string {
+        const DEFAULT_FORMAT = "";
+        if (isEmpty(result)) {
+            return DEFAULT_FORMAT;
+        }
+        const measureGroupHeaderItem = get(result, "dimensions[0].headers[0].measureGroupHeader.items", []);
+        if (measureGroupHeaderItem.length > 0) {
+            return get(measureGroupHeaderItem[indexMeasure], "measureHeaderItem.format", DEFAULT_FORMAT);
+        }
```

Comment:
could it be?
```get(result, `dimensions[0].headers[0].measureGroupHeader.items[${indexMeasure}].measureHeaderItem.format`, DEFAULT_FORMAT);```
:))

Reply:
Yes, it could be just as below and we don't need default value at all
```
private getFormatFromExecutionResponse(indexMeasure: number, result: Execution.IExecutionResponse): string {
    return get(result, `dimensions[0].headers[0].measureGroupHeader.items[${indexMeasure}].measureHeaderItem.format`);
}
```

As I check [here](https://github.com/gooddata/gdc-numberjs/blob/develop/lib/number.js#L78), numberFormat can use it default value.
But please help me re-verify it in runtime.